### PR TITLE
Add new MANAGE_IDENTITIES permission

### LINKED ISSUE #1018: Regular environment users cannot view identities
To reproduce: 

 1. Create an organisation user that has 'VIEW_PROJECT' and 'VIEW_ENVIRONMENT' permissions on a given project / environment
 2. As this user, click on the 'users' option from the left hand menu
 3. Click on one of the users

Expected behaviour: TBD
Actual behaviour: Loading spinner shows and page never loads. 403 in console from attempting following request: 

```
GET https://api.flagsmith.com/api/v1/environments/:key/identities/:id/
```

### PR DESCRIPTION
**Pull request by matthewelwell**
Add a new MANAGE_IDENTITIES permission to allow regular environment users to view the environment detail page. 

Note that as part of this work, I've also tidied up the traits endpoints so that the FE can use those instead of the SDK endpoints as it currently does.

### CODE REVIEW

File: api/environments/identities/traits/views.py [hunk 1]
```diff
@@ -145,21 +88,16 @@ def partial_update(self, request, *args, **kwargs):
...
     )
     def destroy(self, request, *args, **kwargs):
-        delete_all_traits = request.query_params.get("deleteAllMatchingTraits")
-        if delete_all_traits and delete_all_traits in ("true", "True"):
+        if request.query_params.get("deleteAllMatchingTraits") in ("true", "True"):
             trait = self.get_object()
-            self._delete_all_traits_matching_key(
-                trait.trait_key, trait.identity.environment
-            )
+            Trait.objects.filter(
```
Comment (gagantrivedi) [inline api/environments/identities/traits/views.py:94]: @matthewelwell  I don't think we are planning to support this for edge identities?

Comment (matthewelwell) [reply api/environments/identities/traits/views.py:94]: Yeah, that's ok - I don't think it's used tbh.

File: api/environments/identities/views.py [hunk 2]
```diff
@@ -132,6 +133,14 @@ def get_queryset(self):
 
         return queryset
 
+    def get_permissions(self):
```
Comment (gagantrivedi) [inline api/environments/identities/views.py:55]: This line(https://github.com/Flagsmith/flagsmith/pull/1029/files#diff-55e28274d29ce084198b7aacadcd41d63b764f7adea2d1be9d6b15f19da96907L111) is only used for  documentation now, right?

**Review by gagantrivedi — CHANGES_REQUESTED** [review]
I think we need to do the same for edge identity and traits view set, right? Happy to chip in tho

**Comment by matthewelwell** [comment]
> I think we need to do the same for edge identity and traits view set, right? Happy to chip in tho

Yes, good point. Should be pretty simple to add now that those viewsets are merged. I will do that now.

File: api/environments/identities/views.py (hunk 2 above)
Comment (matthewelwell) [reply api/environments/identities/views.py:55]: That link doesn't seem to take me to the correct line but I imagine you were trying to link to the `permission_classes` attribute on the view? I think I've removed that now.

**Review by gagantrivedi — APPROVED** [review]

**Pull request merged**